daily_milestone.go

```diff
@@ -166,21 +166,19 @@ func createMilestoneData(advance int, milestoneTime string) []simpleMilestone {
 			milestone.DueDate = date
 			list = append(list, milestone)
 		} else if milestoneTime == "weekly" {
-
 			today = today.AddDate(0, 0, 7)
```

Comment:
So for weekly that not always works. At least not for the dueDate.
Imagine running dailymile on Tuesday. This would add 7 days. So the current week wouldn't be created and the dueDate would be on a Tuesday even so it should be set to the last day of the week.
Also by just adding 7 days you would only ever add one week milestone as you are not actually iterating over the advance.
So today = today.AddDate(0, 0, i+7) could be fine, but needs a bit more code to get the current week (mostly done) and the last day of a week (might be a bit more complicated and could be done as separate function.

The solution is a bit more difficult. Try to figure out the logic necessary.